File: extract_pdf.js
const fs = require('fs');
const path = require('path');

// We need to use pdfjs-dist for Node.js
const pdfjsLib = require('pdfjs-dist/legacy/build/pdf.js');

async function extractPdfText(pdfPath) {
    const data = new Uint8Array(fs.readFileSync(pdfPath));
    const pdf = await pdfjsLib.getDocument({ data }).promise;
    
    let fullText = '';
    const pagesData = [];
    
    for (let i = 1; i <= pdf.numPages; i++) {
        const page = await pdf.getPage(i);
        const content = await page.getTextContent();
        
        // Sort items by Y position (top to bottom) then X (left to right)
        const items = content.items.sort((a, b) => {
            const yd = b.transform[5] - a.transform[5];
            return yd !== 0 ? Math.sign(yd) : a.transform[4] - b.transform[4];
        });
        
        let pageText = '';
        let lastY = null, lastX = null;
        
        for (const item of items) {
            const x = item.transform[4];
            const y = Math.round(item.transform[5]);
            
            if (lastY !== null && Math.abs(y - lastY) > 3) {
                pageText += '\n';
            } else if (lastX !== null && (x - lastX) > 4) {
                pageText += ' ';
            }
            
            pageText += item.str;
            lastX = x + (item.width || item.str.length * 6);
            lastY = y;
        }
        
        pagesData.push({
            pageNumber: i,
            text: pageText,
            rawItems: content.items.map(item => ({
                str: item.str,
                transform: item.transform,
                fontName: item.fontName,
                width: item.width,
                height: item.height
            }))
        });
        
        fullText += `\n=== PAGE ${i} ===\n` + pageText + '\n';
    }
    
    return {
        fileName: path.basename(pdfPath),
        numPages: pdf.numPages,
        fullText: fullText,
        pages: pagesData
    };
}

const pdfPath = path.join(__dirname, 'PCPD_Borba_123-26.pdf');

extractPdfText(pdfPath)
    .then(result => {
        const outputPath = path.join(__dirname, 'PCPD_Borba_123-26_extracted.json');
        fs.writeFileSync(outputPath, JSON.stringify(result, null, 2));
        console.log('Extração concluída! Arquivo salvo em:', outputPath);
        console.log('\n=== TEXTO COMPLETO ===');
        console.log(result.fullText);
    })
    .catch(err => {
        console.error('Erro:', err);
    });

File: extract_pdf.py
import fitz  # PyMuPDF
import json
import sys

def extract_pdf(pdf_path):
    doc = fitz.open(pdf_path)
    result = {
        "fileName": pdf_path.split('\\')[-1],
        "numPages": len(doc),
        "fullText": "",
        "pages": []
    }
    
    for i, page in enumerate(doc):
        # Get text with position info
        blocks = page.get_text("dict")["blocks"]
        page_text = ""
        
        for block in blocks:
            if "lines" in block:
                for line in block["lines"]:
                    line_text = ""
                    for span in line["spans"]:
                        line_text += span["text"]
                    page_text += line_text + "\n"
        
        result["pages"].append({
            "pageNumber": i + 1,
            "text": page_text.strip()
        })
        result["fullText"] += f"\n=== PAGE {i+1} ===\n{page_text}\n"
    
    doc.close()
    return result

if __name__ == "__main__":
    pdf_path = r"H:\Meu Drive\Inteligência Artificial\INTELIGÊNCIA ARTIFICIAL\STA\Sistema_STA\PCPD_Borba_123-26.pdf"
    
    try:
        import fitz
    except ImportError:
        print("PyMuPDF não instalado. Instalando...")
        import subprocess
        subprocess.check_call([sys.executable, "-m", "pip", "install", "pymupdf"])
        import fitz
    
    result = extract_pdf(pdf_path)
    
    output_path = r"H:\Meu Drive\Inteligência Artificial\INTELIGÊNCIA ARTIFICIAL\STA\Sistema_STA\PCPD_Borba_123-26_extracted.json"
    with open(output_path, 'w', encoding='utf-8') as f:
        json.dump(result, f, ensure_ascii=False, indent=2)
    
    print(f"Extração concluída! Arquivo salvo em: {output_path}")
    print("\n=== TEXTO COMPLETO ===")
    print(result["fullText"])